src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs

```diff
@@ -0,0 +1,88 @@
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            CosmosDbMetadata cosmosDbMetadata = JsonConvert.DeserializeObject<CosmosDbMetadata>(triggerMetadata.Metadata.ToString());
+            cosmosDbMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            IOptions<CosmosClientOptions> options = serviceProvider.GetService<IOptions<CosmosClientOptions>>();
+            ICosmosDBServiceFactory serviceFactory = serviceProvider.GetService<ICosmosDBServiceFactory>();
+            CosmosClient cosmosClient = serviceFactory.CreateService(cosmosDbMetadata.Connection, options.Value);
+            var monitoredContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, cosmosDbMetadata.ContainerName);
+            var leaseContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, string.IsNullOrEmpty(cosmosDbMetadata.LeaseContainerName) ? CosmosDBTriggerConstants.DefaultLeaseCollectionName : cosmosDbMetadata.LeaseContainerName);
+            _scaleMonitor = new CosmosDBScaleMonitor(triggerMetadata.FunctionName, loggerFactory.CreateLogger<CosmosDBScaleMonitor>(), monitoredContainer, leaseContainer, cosmosDbMetadata.LeaseContainerPrefix);
+            _targetScaler = new CosmosDBTargetScaler(triggerMetadata.FunctionName, cosmosDbMetadata.MaxItemsPerInvocation, monitoredContainer, leaseContainer, cosmosDbMetadata.LeaseContainerPrefix, loggerFactory.CreateLogger<CosmosDBTargetScaler>());
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            return _scaleMonitor;
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class CosmosDbMetadata
+        {
+            [JsonProperty]
+            public string Connection { get; set; }
+
+            [JsonProperty]
+            public string DatabaseName { get; set; }
+
+            [JsonProperty]
+            public string ContainerName { get; set; }
+
+            [JsonProperty]
+            public string LeaseContainerName { get; set; }
+
+            [JsonProperty]
+            public string LeaseContainerPrefix { get; set; }
+
+            [JsonProperty]
+            public int MaxItemsPerInvocation { get; set; }
+
+            public void ResolveProperties(INameResolver resolver)
+            {
+                if (resolver != null)
+                {
+                    DatabaseName = resolver.ResolveWholeString(DatabaseName);
+                    ContainerName = resolver.ResolveWholeString(ContainerName);
+                    LeaseContainerName = resolver.ResolveWholeString(LeaseContainerName);
```

Comment:
LeaseContainerName is optional. If the user is not setting it, we use "leases" as default value -> https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerConstants.cs#L8